### ISSUE TITLE
translateX and translateY related issues 

### ISSUE THREAD

**Issue by JerryJohnThomas** [nandorojo/moti#83]
```import React, { useReducer } from 'react'
import { StyleSheet, Pressable } from 'react-native'

import { MotiView, AnimatePresence } from 'moti'


function Shape2() {
    return (
      <MotiView
        from={{
          opacity: 0.5,
          scale:0.5,
          translateY:-200,
          translateX:200,
        }}
        animate={{
          opacity: 1,
          scale:1,
          translateY:0,
          translateX:0,
        }}


        style={styles.shape}
      />
    )
  }

export default function App() {
  const [visible, toggle] = useReducer((s) => !s, true)

  return (
    <Pressable onPress={toggle} style={styles.container}>
      <AnimatePresence>{visible && <Shape2 />}</AnimatePresence>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  shape: {
    justifyContent: 'center',
    height: 250,
    width: 250,
    borderRadius: 25,
    marginRight: 10,
    backgroundColor: 'white',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    backgroundColor: '#9c1aff',
  },
})
```

When having translateX and translateY together it acts weird, u can only see the translateY happening, it seems translateX is happening before that as a sequence and not parallely. I am not that sure but also the scale thing also freezes when u use translateX and Y with exit and all, any insight into this would be helpful

**Comment by nandorojo** [comment]
I'd have to look into this. Might be related to #74

**Comment by power-f-GOD** [comment]
Yeah, I noticed same. 🤦🏽‍♀️  Using translateY and translateX... translateX works, but Y doesn't. It just jumps to its final value, 0:

```
    ... 
    <View style={S.Login}>
      <MotiView
        style={S.logoWrapper}
        from={{
          scale: 1.5,
          translateX: (Dimensions.get('window').width - padding.md * 1.25) / 2,
          translateY: (Dimensions.get('window').height - padding.md * 8) / 2
        }}
        animate={{
          scale: 1,
          translateX: 0,
          translateY: 0
        }}
        transition={{
          type: 'timing',
          duration: 2000,
          delay: 2000
        }}>
        <Logo
          style={
            S.logo
          }
        />
      </MotiView>
    </View>
    ...
```

Any solutions to this yet, @nandorojo

**Comment by power-f-GOD** [comment]
Actually, I think using either of the CSS transform properties together will make only one of the transforms work.
I mean, if you use either of `scale` and `translateX` or `scale` and `translateY` or `translateY` and `translateX` or `rotate` and `scale` etc., only one of the properties end up animating; the others don't.

**Comment by nandorojo** [comment]
I think this is a duplicate of #76. I wonder if the object keys aren't in proper order...

**Comment by nandorojo** [comment]
I know this isn't ideal, but for now, could you use 2 moti views, and apply only one transform per component? Until I have time to look more closely at this.

**Comment by nandorojo** [comment]
Temporary solution for multiple transforms can be found here: https://github.com/nandorojo/moti/issues/85#issuecomment-895599767

TLDR: use `useAnimationState` instead of the `from` and `animate` props until I can investigate the proper cause of this.

**Comment by nandorojo** [comment]
Would be great if you guys could test there so I know if it's good to go.

**Comment by nandorojo** [comment]
Okay I found the cause of this. Reanimated is changing the order of the keys. It seems that the object key order is not reliable for transform.

**Comment by nandorojo** [comment]
https://github.com/software-mansion/react-native-reanimated/issues/2352

Issue there. I found the solution, though. Will be merged soon.

**Comment by nandorojo** [comment]
Fixed in `0.12.0`

```sh
yarn add moti
```
